VirtoCommerce.MarketingModule.Web/Scripts/promotion/blades/promotion-expressions.js

```diff
@@ -22,52 +22,165 @@ angular.module('virtoCommerce.marketingModule')
             toolbarCommands: [
             {
-                name: "platform.commands.pick-selected", icon: 'fa fa-plus',
+                name: "platform.commands.refresh", icon: 'fa fa-refresh',
                 executeMethod: function (blade) {
-                    //parentElement.selectedListEntry = selectedListEntries[0];
-                    parentElement.productId = selectedListEntries[0].id;
-                    parentElement.productName = selectedListEntries[0].name;
-                    parentElement.productCode = selectedListEntries[0].code;
-                    bladeNavigationService.closeBlade(blade);
+                    blade.refresh();
+                },
+                canExecuteMethod: function () { return true; }
+            },
+            {
+                name: "platform.commands.add", icon: 'fa fa-plus',
+                executeMethod: function (blade) {
+                    openCatalogBlade(blade, newBlade);
                 },
                 canExecuteMethod: function () {
-                    return selectedListEntries.length == 1;
+                    return true;
                 }
-            }]
-        };
+            },
+            {
+                name: "platform.commands.delete", icon: 'fa fa-trash-o',
+                executeMethod: function (blade) {
+                    var grid = blade.$scope.gridApi.grid;
+                    var selectedRows = grid.api.selection.getSelectedRows();
+                    for (var entry of selectedRows) {
+                        grid.options.data = _.filter(grid.options.data, function (item) {
+                            return item.id != entry.id;
+                        });
+                    }
 
-        newBlade.options = {
-            showCheckingMultiple: false,
-            checkItemFn: function (listItem, isSelected) {
-                if (listItem.type == 'category') {
-                    newBlade.error = 'Must select Product';
-                    listItem.selected = undefined;
-                } else {
-                    if (isSelected) {
-                        if (_.all(selectedListEntries, function (x) { return x.id != listItem.id; })) {
-                            selectedListEntries.push(listItem);
-                        }
+                    blade.selectedListEntries = grid.options.data.slice();
+                    parentElement.productIds = _.map(grid.options.data, function (item) {
+                        return item.id;
+                    });
+                    blade.productIds = parentElement.productIds.slice();
+                },
+                canExecuteMethod: function (blade) {
+                    if (blade.childrenBlades.length > 0) {
+                        return false;
                     }
-                    else {
-                        selectedListEntries = _.reject(selectedListEntries, function (x) { return x.id == listItem.id; });
+                    if (blade.$scope.gridApi) {
```

Comment:
Probably better to make all work with grid in blade were grid is located and created.